src/firefly/js/ui/CatalogSearchMethodType.jsx

```diff
@@ -0,0 +1,262 @@
+                                 initialState={{
+                                          fieldKey: 'axialratio',
+                                          value: '.26',
+                                          validator: Validate.floatRange.bind(null, 0, 1, 0,'Axial Ratio'),
+                                          tooltip: 'Enter the Axial ratio of the search e.g - 0.26',
+                                          label : 'Axial Ratio:',
+                                          labelWidth : 100
+                                      }}/>
+            </div>
+        );
+    } else if (searchType === SpatialMethod.Box.value) {
+
+        return (
+            <div className='size'>
+                {radiusInField('Side:', 'Enter side size of the box search', 1 / 3600, 7200 / 3600)}
+            </div>
+
+        );
+    } else if (searchType === SpatialMethod.get('Multi-object').value) {
+        return (
+            <div style={{padding:5, display:'flex', flexDirection:'column', flexWrap:'no-wrap', alignItems:'center' }}>
+                <FileUpload
+                    wrapperStyle={{margin: '5px 0'}}
+                    fieldKey='fileUpload'
+                    initialState={{
+                        tooltip: 'Select a  file to upload',
+                        label: 'Filename:'}}
+                />
+                {radiusInField()}
+            </div>
+        );
+    } else if (searchType === SpatialMethod.Polygon.value) {
+
+        return (
+            <div style={{padding:5, display:'flex', flexDirection:'column', flexWrap:'no-wrap', alignItems:'center' }}>
+                <InputAreaFieldConnected fieldKey='polygoncoords'
+                                wrapperStyle={{display:'flex', flexDirection:'row', alignItems:'center' }}
+                                initialState={{
+                                               tooltip:'Enter polygon coordinates search'
+                                            }}
+                                label='Coordinates:'
+                                tooltip='Enter polygon coordinates search'
+                />
+                <ul>
+                    <li>- Each vertex is defined by a J2000 RA and Dec position pair</li>
+                    <li>- A max of 15 and min of 3 vertices is allowed</li>
+                    <li>- Vertices must be separated by a comma (,)</li>
+                    <li>- Example: 20.7 21.5, 20.5 20.5, 21.5 20.5, 21.5 21.5</li>
+                </ul>
+            </div>
+        );
+
+    } else {
+        return (
+
+            <div className='size'>
+                Search the catalog with no spacial constraints
+            </div>
+        );
+    }
+}
+
+function renderTargetPanel(groupKey, searchType) {
+    const visible = (searchType === SpatialMethod.Cone.value || searchType === SpatialMethod.Box.value || searchType === SpatialMethod.Elliptical.value) && searchType != SpatialMethod.get('Multi-Object').value;
```

Comment:
Correct, the extra check on 'multi-object' option is not necessary. Good 
point.
About improving readability of the code, that can be addressed too in a 
second pass, right now i don't see a better way.

On 5/16/16 2:40 PM, loitly wrote:
>
> In src/firefly/js/ui/CatalogSearchMethodType.jsx 
> <https://github.com/Caltech-IPAC/firefly/pull/85#discussion_r63431067>:
>
> > +                </ul>
> > +            </div>
> > +        );
> > +
> > +    } else {
> > +        return (
> > +
> > +            <div className='size'>
> > +                Search the catalog with no spacial constraints
> > +            </div>
> > +        );
> > +    }
> > +}
> > +
> > +function renderTargetPanel(groupKey, searchType) {
> > +    const visible = (searchType === SpatialMethod.Cone.value || searchType === SpatialMethod.Box.value || searchType === SpatialMethod.Elliptical.value) && searchType != SpatialMethod.get('Multi-Object').value;
>
> I think '&& searchType != SpatialMethod.get('Multi-Object').value' is 
> not necessary.
> Also, there might be another way to do these 'in' comparison that's 
> easier to read.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/85/files/8bdfd737fd8631b45d10aff1ebb2742bee675e9e#r63431067>
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489